src/languageSupport/languages/sql/monaco.sql.syntax.ts

```diff
@@ -0,0 +1,23 @@
+import * as pgsql from 'monaco-editor/esm/vs/basic-languages/pgsql/pgsql'
+
+import {MonacoBasicLanguage} from 'src/types'
+
+const LANGID = 'pgsql'
```

Comment:
Is it required in Monaco editor to have language id to be `pgsql`? I tried to look up Monaco editor docs but couldn't find any. The reason I asked is because there is an enum `LanguageType.SQL` that is set to be `sql`. I'd like to see if it is possible to combine these two constants in one way or another

Reply:
monaco-editor provides a series of language syntax out of the box. We use the `MonacoWebpackPlugin` to determine which languages to load -- specifically, those languages where monaco-editor also provides a baked-in LSP.

```
new MonacoWebpackPlugin({
      languages: ['json', 'markdown'],
      filename: `${STATIC_DIRECTORY}[name].worker.[contenthash].js`,
      globalAPI: true,
    })
```

For other languages, specifically those found under the[ basic-languages directory here](https://github.com/microsoft/monaco-editor/tree/main/src/basic-languages), there is no baked-in LSP available. But there are syntaxes available, which we load and use with our own custom color theme.

In that^^ list of basic syntaxes, monaco-editor uses the language id `pgsql`. Since our backend dialect is postgres sql, I choose that sql dialect for the syntax.